plugins/woocommerce-admin/client/dashboard/components/connect/index.js

```diff
@@ -7,64 +7,65 @@ import { Component, Fragment } from '@wordpress/element';
 import { compose } from '@wordpress/compose';
 import PropTypes from 'prop-types';
 import { withDispatch, withSelect } from '@wordpress/data';
-import { PLUGINS_STORE_NAME } from '@woocommerce/data';
+import { ONBOARDING_STORE_NAME } from '@woocommerce/data';
 
+/**
+ * Button redirecting to Jetpack auth flow.
+ *
+ * Only render this component when the user has accepted Jetpack's Terms of Service.
+ * The API endpoint used by this component sets "jetpack_tos_agreed" to true when
+ * returning the URL.
+ */
 export class Connect extends Component {
 	constructor( props ) {
 		super( props );
 		this.state = {
-			isConnecting: false,
+			isAwaitingRedirect: false,
```

Comment:
Previously, this button would be disabled until it gets the URL. This used to be a quick operation but with the new endpoint, getting the URL from the API takes a while.

I tried resolving it by adding a prefetch operation that would retrieve the URL when the entire stepper is loaded. This worked but since accessing the `jetpack-authorization-url` endpoint also sets `jetpack_tos_accept` to `true` in the database, I decided if we start prefetching immediately after the user accepts the TOS (= which is when we show the Connect button), we still speed things up a bit.

The button is no longer disabled when the URL is being loaded. Instead, it will register the fact that it was clicked, display a loading animation (`isBusy`) and only actually redirect once it gets the URL.